addons/website/static/src/js/widgets/theme.js

```diff
@@ -37,9 +43,159 @@ var ThemeCustomizeDialog = Widget.extend({
             });
         }
         return $.when(this._super.apply(this, arguments), templateDef);
+
     },
     start: function () {
         var self = this;
+
+        // Theme modal
+        var $contents = this.$el.children('content');
+
+        $contents.remove();
+        this.$el.append(QWeb.render('website.modal_custom_theme'));
+        var $navContents = this.$el.find('.o_container_right');
+        var $navString = this.$el.find(".o_el_nav_modal");
+
+        // The big loop, to interpret xml
+        _.each($contents, function (content) {
+            var $content = $(content);
+
+            // Build the nav of the panel
+            $navString.append(QWeb.render('website.nav_theme_modal', {
+                string: $content.data('string'),
+                href: "#" + $content.data('id'),
+                classadd: $content.data('classactive'),
+                id: $content.data('id'),
+            }));
+
+            // Build the main of the panel
+            var $navContent = $(QWeb.render('website.multi_choice', {
+                title: $content.data('title'),
+                id: $content.data('id'),
+                class: $content.data('class'),
+            }))
+            $navContents.append($navContent);
+
+            var $optionBtn = $navContent.find('.o_initial_list');
+            var $options = $content.children('opt');
+
+            // Build the options template
+            _.each($options, function (option) {
+                var $option = $(option);
+
+                $optionBtn.append(QWeb.render('website.multi_choice_content', {
+                    strings: $option.data('strings'),
+                    xmlid: $option.data('xmlid'),
+                    class: $option.data('class'),
+                    liclass: $option.data('liclass'),
+                }));
+
+                var $paletColor = $optionBtn.find('.o_add_color_choice');
+                var $paletContainer = $paletColor.parent('ul');
+
+                $paletContainer.append(QWeb.render('website.add_color_choice'));
+
+                var $paletContent= $paletContainer.find('.o_choice_boxed_background');
+                $paletContent.append(QWeb.render('website.actually_theme_color'));
+                $paletContent.append(QWeb.render('website.palet_color'));
+                $paletContent.append(QWeb.render('website.pattern'));
+
+                var $patternList = $paletContent.find('.o_pattern_list');
+                $patternList.append(QWeb.render('website.pattern_element'));
+
+                $paletContent.append(QWeb.render('website.add_photo'));
```

Comment:
The whole color part should be handled in another way but we will see that later. Maybe even after we merge this task. The goal of the current task is mainly to improve the modal *generation* and give it an Odoo-like style. Improving the options themselves can be done in a second time. I suggest you remove that part of your current work and save it for later.